packages/object/computed.d.ts

```diff
@@ -340,6 +340,70 @@ export function lt(dependentKey: string, value: Number): PropertyDecorator;
  * @param {Number} value - Value to compare against
  */
 export function lte(dependentKey: string, value: Number): PropertyDecorator;
+/**
+  Creates a new decorator from a computed macro function. For instance, you can
+  use this utility function to create decorators from the macros provided by
+  addons such as [ember-awesome-macros](https://github.com/kellyselden/ember-awesome-macros).
+
+  ```js
+  import { macro } from '@ember-decorators/object/computed';
+  import firstMacro from 'ember-awesome-macros/array/first';
+
+  const first = macro(firstMacro);
+
+  export default class LeaderBoardComponent extends Component {
+    ranking = ['Natalie', 'Emma', 'Thomas'];
+
+    @first('ranking') winner; // => 'Natalie'
+
+  }
+  ```
+
+  You can also make use of [partial application](http://2ality.com/2011/09/currying-vs-part-eval.html)
+  of arguments:
+
+  ```js
+  import { macro } from '@ember-decorators/object/computed';
+  import { computed } from '@ember/object';
+
+  const titleGeneratorMacro = (prefix, titleKey) => computed(function() {
+    return `${prefix}: ${String(get(this, titleKey)).toUpperCase()}!`;
+  });
+
+  const newsFlash = macro(titleGeneratorMacro, 'News Flash');
+
+  export default class NewsPaperComponent extends Component {
+    title = 'Ember chosen best framework of the year again';
+
+    @newsFlash('title') attentionGrabber; // => 'News Flash: EMBER CHOSEN BEST FRAMEWORK OF THE YEAR AGAIN!'
+  }
+  ```
+
+  @param {function} fn - The macro function to create a decorator from
```

Comment:
I _think_ this also needs an `@function` like the other doc blocks.

Reply:
Would a `@function` tag be redundant in `computed.d.ts`? I couldn't actually see any mention of this tag in TSDoc, but may have missed it.

Looking at `computed.js`, the `@function` tags were added to exports that weren't self-evidently functions, but `macro()` is defined as a function directly in the source file, so again it may be redundant.

Just not 100% sure of the documentation semantics here.

Reply:
> Looking at computed.js, the @function tags were added to exports that weren't self-evidently functions,

Yes, that's why I did not add the tag there.

> but macro() is defined as a function directly in the source file, so again it may be redundant.

If TS is smart enough to deduce this, I guess we can skip it here. By that logic, we can probably skip it in all of `computed.d.ts` then. 🤔

Reply:
Assuming this is a copy-paste job it's probably easier to just keep it the same in both?

Reply:
I don't think it matters much, since we aren't using the output of TypeDoc yet. We can consider switching over in the future since it'll be able to infer even more.